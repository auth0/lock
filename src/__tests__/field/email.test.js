import Immutable from 'immutable';

describe('field/email', () => {
  let email;
  beforeEach(() => {
    jest.resetModules();

    jest.mock('field/index', () => ({
      setField: jest.fn()
    }));

    email = require('field/email');
  });
  describe('setEmail()', () => {
    it(`trims email`, () => {
      email.setEmail(Immutable.fromJS({}), ' email@test.com ');
      const { mock } = require('field/index').setField;
      expect(mock.calls.length).toBe(1);
      expect(mock.calls[0]).toMatchSnapshot();
    });
  });
});
