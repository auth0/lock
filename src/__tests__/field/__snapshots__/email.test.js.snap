// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`field/email setEmail() trims email 1`] = `
Array [
  Immutable.Map {},
  "email",
  "email@test.com",
  [Function],
]
`;
